### PR TITLE
remove redundant ajax call when select in file explorer

### PR DESCRIPTION
**Pull request by watashiSHUN**
current behavior when you click in file explorer:
capture
new behavior:
image

@ahmelsayed 
while I am working on the front end routing...
I would suggest that we move the logic to **notify command line from explorer** out of the [selected observable.subscribe()](https://github.com/projectkudu/kudu/blob/master/Kudu.Services.Web/Content/Scripts/FileBrowser.js#L519-L528) (asking for approval :))

for file browser functions, we can notify command line explicitly, for example
`notify command line; selected(new value)`

and when command line wants to notify the file explorer about directory change, we just do
`select(new value)`

so we don't need the [flag variable](https://github.com/projectkudu/kudu/blob/master/Kudu.Services.Web/Content/Scripts/FileBrowser.js#L502) and the [extra promise](https://github.com/projectkudu/kudu/blob/master/Kudu.Services.Web/Content/Scripts/FileBrowser.js#L333)...and make the code readable

### CODE REVIEW

**Comment by davidebbo** [comment]
Not familiar with that part of the code, so I'll let others review this one.

Note: you can go to https://github.com/projectkudu/kudu/pull/2492/files?w=1 to get a much smaller diff that ignores white space changes.

**Comment by ahmelsayed** [comment]
@watashiSHUN I don't have any objections. It'll be great if you can making that code more readable even if just a little bit. the communication between the console and the fileBrowser is messy

File: Kudu.Services.Web/Content/Scripts/FileBrowser.js [hunk 1]
```diff
@@ -368,8 +360,7 @@ $.connection.hub.start().done(function () {
             viewModel.editText('');
             statusbar.fetchingContents();
             viewModel.fileEdit(this);
-            if(this.mime === "text/xml")
-            {
+            if (this.mime === "text/xml") {
                 Vfs.getContent(this)
                    .done(function (data) {
                        viewModel.editText(vkbeautify.xml(data));
```
Comment (snobu) [inline Kudu.Services.Web/Content/Scripts/FileBrowser.js]: Do we still need to vkbeautify XML? I believe the Ace editor is already doing that if the extension is .xml (loads the XML mode). I'll test and confirm. vkbeautify was historically used here just to get a readable XML in the (old) editor, right?

Comment (snobu) [inline Kudu.Services.Web/Content/Scripts/FileBrowser.js]: Nope, i was wrong. Setting XML mode in Ace does not beautify the content, only enables coloring. Let's just leave that in.

**Comment by davidebbo** [comment]
@watashiSHUN go ahead and get this in if you think it's all done.

**Comment by watashiSHUN** [comment]
06e0777b8d9ec2353f1b4fcc1efd89ad45bc2d4e

**Pull request closed without merging**